examples/hello-world/hello_service.py

```diff
@@ -19,30 +19,35 @@ async def get_payload():
         )
 
     async def start(self):
+        async for s in super().start():
+            yield s
+
         # every `DATE_POLL_INTERVAL` write output of `date` to `DATE_OUTPUT_PATH`
-        self._ctx.run(
+        s = self._ctx.new_script()
+        s.run(
             "/bin/sh",
             "-c",
             f"while true; do date > {DATE_OUTPUT_PATH}; sleep {REFRESH_INTERVAL_SEC}; done &",
         )
-        yield self._ctx.commit()
+        yield s
 
     async def run(self):
         while True:
             await asyncio.sleep(REFRESH_INTERVAL_SEC)
-            self._ctx.run(
+            s = self._ctx.new_script()
+            s.run(
                 "/bin/sh",
                 "-c",
                 f"cat {DATE_OUTPUT_PATH}",
             )
 
-            future_results = yield self._ctx.commit()
+            future_results = yield s
             results = await future_results
-            print(results[0].stdout.strip())
+            print(results[0].stdout.strip() if results[0].stdout else "")
```

Comment:
Similarly to `hello-world/hello.py` above, I think we can use the result from the `run` command directly instead of the array of all results.